### DISCUSSION TITLE
How can I deploy Lowdefy apps with version 4.0.0-rc.5 using AWS Lambda (Serverless)?

### DISCUSSION THREAD

**Discussion by vicam001** [lowdefy/lowdefy#1518] [Questions and Help]
Hi,
I check the Lowdefy docs for version 4.0.0-rc.5 ( https://docs-git-version-lowdefy.vercel.app) and I cannot see the option to deploy using AWS Lambda (Serverless) that was available in version 3.23.3 ( https://docs.lowdefy.com/aws-lambda ) 

This was one of the reasons for choosing Lowdefy and right now I am struggling to get deployed a new app that I have upgraded from v3.23.3 to v4.0.0-rc.5.

I am trying to achieve something similar with AWS Amplify but is by far more complex and not as straight forward as with the Serverless framework in v3.23.3.

Can you please share some thoughts on this?

Apparently could be solved by publishing into DockerHub a new image, instead of lowdefy/lowdefy-aws-lambda:3.23.3, lowdefy/lowdefy-aws-lambda:4.0.0-rc.5, but this is beyond my technical capabilities.

Thank you very much for the support.

**Comment by SamTolmay** [comment]
Hi @vicam001

Unfortunately a Docker image won't work on AWS Lambda, as Lowdefy V4 is a Next.js app.

The `lowdefy build` command generates the Next app, which is then built to package all in all the plugins used in the app. 

I tried to set up an AWS Amplify example, but unfortunately haven't been able to get it to work. Everything looks like it installs and builds correctly, but it looks like Amplify isn't picking up that I am trying to deploy a Next app. When visiting the deployed site I get a 404. You can look at what I have so far here, maybe you can get it to work: https://github.com/lowdefy/lowdefy-example-aws-amplify

**Reply by vicam001** [reply]
Thanks @SamTolmay for the explanation.

I have just created the PR https://github.com/lowdefy/lowdefy-example-aws-amplify/pull/1 with something that makes me be able to deploy my 4.0.0-rc.5 Lowdefy app into AWS.

It works now, what I am struggling  is with the cold start of the app, experiencing very large Times To First Bite (TTFB).

**Reply by Gervwyk** [reply]
Hi @vicam001, even on Vercel our cold start times are terrible (>7 seconds). It's simply a disadvantage of running on Lambda at the moment. To fix this we run a seperate cron service that pings all our apps every two minutes. Not the most elegant solution but it is working.

**Reply by SamTolmay** [reply]
I am not sure how images work on Amplify, but moving them to a CDN or AWS S3 might help cold start times since the bundle is smaller
